tests/integration/pilot/revisions/uninstall_test.go

```diff
@@ -155,43 +100,69 @@ func TestUninstallPurge(t *testing.T) {
 			istioCtl.InvokeOrFail(t, uninstallCmd)
 			cs := ctx.Clusters().Default()
 			checkCPResourcesUninstalled(t, cs, append(helmreconciler.NamespacedResources, helmreconciler.AllClusterResources...),
-				helmreconciler.IstioComponentLabelStr)
+				helmreconciler.IstioComponentLabelStr, true)
 		})
 }
 
 // checkCPResourcesUninstalled is a helper function to check list of gvk resources matched with label are uninstalled
-func checkCPResourcesUninstalled(t *testing.T, cs cluster.Cluster, gvkResources []schema.GroupVersionKind, label string) {
+// If purge is set to true, we expect all resources are removed.
+// Otherwise we expect only selected resources from control plane are removed, resources from base and the legacy addon installation would not be touched.
+func checkCPResourcesUninstalled(t *testing.T, cs cluster.Cluster, gvkResources []schema.GroupVersionKind, label string, purge bool) {
 	retry.UntilSuccessOrFail(t, func() error {
+		var reStrlist []string
+		var reItemList []unstructured.Unstructured
 		for _, gvk := range gvkResources {
 			resources := strings.ToLower(gvk.Kind) + "s"
 			gvr := schema.GroupVersionResource{Group: gvk.Group, Version: gvk.Version, Resource: resources}
-
-			if err := checkResourcesNotInCluster(cs, gvr, label); err != nil {
-				return err
-			}
+			reList, reStr := getRemainingResourcesCluster(cs, gvr, label)
+			reItemList = append(reItemList, reList...)
+			reStrlist = append(reStrlist, reStr...)
 		}
-		return nil
-	}, retry.Delay(time.Millisecond*100), retry.Timeout(time.Second*120))
+		return inspectRemainingResources(reItemList, reStrlist, purge)
+	}, retry.Delay(time.Millisecond*100), retry.Timeout(time.Second*150))
```

Comment:
what's the reason for this timeout increase? we should really move this to a file constant and give it a name.

Reply:
done